### PR TITLE
Fix crash in VerifyForwardedTypes

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Test forthcoming.  It's somewhat complicated because you need to have forwarded types, and multiple assemblies/modules with the same assembly name.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Reviewers: @balajikris, @basoundr, @brettfo, @davkean, @DustinCampbell, @dpoeschl, @Pilchie

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski retest this please

**Comment by CyrusNajmabadi** [comment]
Oops.  Ignore that @jasonmalinowski

File: src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs [hunk 1]
```diff
@@ -338,7 +344,7 @@ protected static bool IdentifiersMatch(ISyntaxFactsService syntaxFacts, string n
                 }
 
                 // 3) If the symbols ARE equivalent ignoring assemblies, they may or may not be equivalent if containing assemblies are NOT ignored.
-                if (equivalentTypesWithDifferingAssemblies.Object.Any())
```
Comment (jaredpar) [inline src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs]: We really should add an analyzer for this.  We keep finding latent instances where we're throwing away perf.

**Comment by CyrusNajmabadi** [comment]
@jaredpar I agree.  If i have some time between things, i'll try to whip something up!

**Comment by Pilchie** [comment]
Looks good, but waiting for aforementioned tests.

**Comment by jasonmalinowski** [comment]
Just checking -- did we ever get tests for this?

**Pull request merged**